axelor-contract/src/main/resources/views/ConsumptionLine.xml

```diff
@@ -70,10 +70,14 @@
 
   <action-attrs name="action-consumption-line-attrs-set-product-domain">
     <attribute name="domain" for="product"
-      expr="eval: &quot; self.dtype = 'Product' AND self.sellable = true &quot;"
+      expr="eval: __parent__?.company &amp;&amp; __config__.app.getApp('base')?.companySpecificProductFieldsSet?.find({it.name == 'sellable'}) ?
+      &quot; self.dtype = 'Product' AND EXISTS (SELECT pc FROM self.productCompanyList pc WHERE pc.company = &quot; + __parent__.company.id + &quot; AND pc.sellable = true) &quot; :
```

Comment:
Same as the previous comment, this query won't work anymore with AOP 8.0. The company should be compared with a company. Or its id with a long value.